mysql-test/include/ensure_binlog_row_event_columns.inc

```diff
@@ -0,0 +1,67 @@
+#
+# Helper file to ensure that a binary log file properly writes the expected
+# fields based on the binlog_row_image value.
+#
+# ==== Usage ====
+#
+# --let $expected_columns= (COLUMN_IDS)
+# --let $num_columns= NUM
+# --let $binlog_filename= FILENAME
+# --source include/count_binlog_row_event_columns.inc
+#
+# Parameters:
+#   expected_columns (list<uint>, in) : A list of positive integers which
+#       correspond to the column numbers that should be output in a binary
+#       log's write_rows event
+#   num_columns (uint, in) : The number of columns to check for a write_rows
```

Comment:
I am not sure about this parameter.
I read in below that it simply limits/optimizes the comparison loop.
But the loop limit can be also found with `SELECT GREATEST`..

More to that, having the limit this commit's way does not guarantee absence of more fields
with higher indices. And we there'd be, say, the @9th fields in the output your tests would not catch that.
If `$num_columns`  is specified explicitly the loop would stop at reaching the limit, otherwise 
`SELECT 1+GREATEST $expected_columns` column would be also tried (to not be found in the good case).